File: VB-Examples/09_Charts/AddPictureInChart.vb
Imports System.Data.OleDb
Imports System.Collections
Imports System.ComponentModel

Imports Spire.Xls
Imports Spire.Xls.Charts

Namespace AddPictureInChart
	Partial Public Class Form1
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
            ' Create a new Workbook object to represent an Excel workbook
            Dim workbook As New Workbook()

            ' Load an existing Excel file named "ChartToImage.xlsx" from the specified path
            workbook.LoadFromFile("..\..\..\..\..\..\Data\ChartToImage.xlsx")

            ' Get the first worksheet (index 0) from the workbook
            Dim sheet As Worksheet = workbook.Worksheets(0)

            ' Get the first chart (index 0) on the worksheet
            Dim chart As Chart = sheet.Charts(0)

            ' Add a picture to the chart using the specified image file path
            chart.Shapes.AddPicture("..\..\..\..\..\..\Data\SpireXls.png")

            ' Specify the output filename for the modified workbook
            Dim result As String = "result.xlsx"

            ' Save the modified workbook to a new Excel file with the Excel 2010 format
            workbook.SaveToFile(result, ExcelVersion.Version2010)
            ' Release the resources used by the workbook
            workbook.Dispose()

            ExcelDocViewer(result)
		End Sub
		Private Sub ExcelDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

		Private Sub btnClose_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnClose.Click
			Close()
		End Sub
	End Class


End Namespace

File: CS-Examples/09_Charts/AddPictureInChart.cs
using System;
using System.Windows.Forms;
using Spire.Xls;


namespace AddPictureInChart
{
    public partial class Form1 : Form
    {

        public Form1()
        {
            InitializeComponent();
        }

        private void btnRun_Click(object sender, System.EventArgs e)
        {
            // Create a workbook
            Workbook workbook = new Workbook();

            // Load the document from disk
            workbook.LoadFromFile(@"..\..\..\..\..\..\Data\ChartToImage.xlsx");

            // Get the first sheet
            Worksheet sheet = workbook.Worksheets[0];

            // Get the first chart
            Chart chart = sheet.Charts[0];

            // Add the picture in chart
            chart.Shapes.AddPicture(@"..\..\..\..\..\..\Data\SpireXls.png");

            // Save the result file
            string result = "result.xlsx";
            workbook.SaveToFile(result, ExcelVersion.Version2010);

            // Dispose of the workbook object to release resources
            workbook.Dispose();

            // Launch the file
            ExcelDocViewer(result);
        }
        private void ExcelDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

        private void btnClose_Click(object sender, EventArgs e)
        {
            Close();
        }
    }


}
